apps/web/src/pages/Profile/Profile.tsx

```diff
@@ -4,75 +4,153 @@ import { useForm, Controller } from 'react-hook-form';
 import { zodResolver } from '@hookform/resolvers/zod';
 import { z } from 'zod';
 import { FileInput } from '@mantine/core';
+import { notifications } from '@mantine/notifications';
 import styles from './Profile.module.css';
 import Button from '@/components/Button';
 import Input from '@/components/Input';
 import ProfilePicture from '@/assets/svg/profile.svg?react';
+import { useUploadImage } from '@/services/users';
 
 const generalSettingsSchema = z.object({
-  avatar: z.instanceof(FileList),
   name: z.string(),
   country: z.string(),
   address1: z.string(),
   address2: z.string(),
   zip: z.number(),
 });
 
+const MAX_FILE_SIZE = 5000000;
+const ACCEPTED_IMAGE_TYPES = [
+  'image/jpeg',
+  'image/jpg',
+  'image/png',
+  'image/webp',
+];
+
+const inputImageSchema = z.object({
+  image: z
+    .instanceof(File)
+    .refine((file) => file.size <= MAX_FILE_SIZE, `Max image size is 5MB.`)
+    .refine(
+      (file) => ACCEPTED_IMAGE_TYPES.includes(file.type),
+      'Only .jpg, .jpeg, .png and .webp formats are supported.'
+    ),
+});
+
 export type BookingSchemaType = z.infer<typeof generalSettingsSchema>;
+export type InputImageSchemaType = z.infer<typeof inputImageSchema>;
+
+interface ImageFormProps {
+  onUploadImage: (file: File) => void;
+}
+
+function ImageForm({ onUploadImage }: ImageFormProps) {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm<InputImageSchemaType>({
+    resolver: zodResolver(inputImageSchema),
+  });
+
+  const imageValue = watch('image') as File | undefined;
+
+  const onSubmit: SubmitHandler<InputImageSchemaType> = (data) => {
+    const file = data.image
+    onUploadImage(file)
```

Comment:
no es necesario crear la variable aqui
```suggestion
    onUploadImage(data.image)
```